core/MatomoInstaller.php

```diff
@@ -0,0 +1,416 @@
+use Piwik\Url;
+use Piwik\Version;
+use Zend_Db_Adapter_Exception;
+
+//Warning: - Don't know how plugin are installed (not done by installer
+//           actually). Some magics when I created installer.
+//
+// require : https://github.com/matomo-org/matomo/pull/15691
+
+/*
+ * Manage the Matomo server installation. Contain
+ * all basic function to setup database, user, and
+ * site to track.
+ *
+ * (?Could be a parent class to create custom
+ * installation process?)
+*/
+
+class MatomoInstaller{
+    /*
+     * Perform the entire installation of Matomo.
+     * Use user-defined settings.
+     *
+     * @settings is expected to be an array with following
+     * variables.
+     *
+     *  - dbname
+     *  - port (optionnal)
+     *  - dbusername
+     *  - dbpassword
+     *  - dbhost (optionnal?)
+     *  - tables_prefix (optionnal)
+     *  - adapter (optionnal)
+     *  - type (optionnal)
+     *  - adminusername
+     *  - adminpassword
+     *  - email
+     *  - subscribe_newsletter_piwikorg (optionnal)
+     *  - subscribe_newsletter_professionalservices (optionnal)
+     *  - name
+     *  - url
+     *  - ecommerce : (optionnal)
+    */
+    public static function headlessInstall($settings)
+    {
+        if (self::systemCheck())
+        {
+            //Raise Error because DigntosicService report in invalid.
+            //What should I do in case of warning or error ?
+        }
+        
+        self::createDatabaseObject($settings);
+        self::tablesCreation();
+        self::setupSuperUser($settings);
+        self::firstWebsiteSetup($settings);
+    }
+
+    /*
+     * Check if an headless installation is expected.
+     * (Heritance should be related to it)
+     */
+    public static function isHeadlessInstall()
+    {
+        $installSettings = Config::getInstance()->MatomoInstall;
+        return !empty($installSettings);
+    }
+
+    /*
+     * Use settings from MatomoInstall section in
+     * the config.ini.php file and perform installation.
+     */
+    public static function installFromConfig()
+    {
+        //Use settings defined in config.ini.php
+        //in MatomoInstall section.
+        $settings = Config::getInstance()->MatomoInstall;
+        self::headlessInstall($settings);
+
+        //Remove installation config when finished
+        Config::getInstance()->MatomoInstall = array();
+        Config::getInstance()->forceSave();
+
+        //Redirect to matomo homepage
+        Url::redirectToUrl('index.php');
+    }
+
+    /**
+     * Installation Step 2: System Check
+     *
+     * Control if the system is ready for installation,
+     * run Plugins\Diagnostics to get state.
+     * 
+     * @view View (optionnal), view to store DiagnosticReport
+    */
+    //S: Should I perform the system check ?
+    //In : https://github.com/digitalist-se/extratools/blob/master/Lib/Install.php
+    // no system check seem to be performed.
+    public static function systemCheck($view=null)
```

Comment:
When moved to a command, we could output the error results and maybe ask the user if the installation should be aborted or continued nevertheless.